isis/tests/FunctionalTestsCubeatt.cpp

```diff
@@ -0,0 +1,71 @@
+#include <iostream>
+
+#include "Cube.h"
+#include "Fixtures.h"
+
+#include "cubeatt.h"
+#include "gmock/gmock.h"
+
+
+using namespace Isis;
+
+static QString APP_XML = FileName("$ISISROOT/bin/xml/cubeatt.xml").expanded();
+
+// Tests setting output attributes: bit type and range
+TEST_F(SmallCube, FunctionalTestCubeattBitttypeAndRange) {
+  QString cubePath = tempDir.path() + "/bitTypeCubeatt.cub+8bit+0.0:1.0";
+
+  QVector<QString> args = {"from=" + testCube->fileName(), "to=" + cubePath};
+  UserInterface options(APP_XML, args);
+  cubeatt(options);
+
+  // Test the label
+  Cube outputCube(cubePath);
+  Pvl *label = outputCube.label();
+  PvlGroup &pixelGroup = label->findObject("IsisCube").findObject("Core").findGroup("Pixels");
+  EXPECT_EQ(QString(pixelGroup["Type"]).toStdString(), "UnsignedByte");
+  EXPECT_NEAR(pixelGroup["Base"][0].toDouble(), -0.003952569, 0.00001);
+  EXPECT_NEAR(pixelGroup["Multiplier"][0].toDouble(), 0.003952569, 0.00001);
```

Comment:
Use the Cube accessors for these instead of text parsing the label. Also, I would not check the base and multiplier, just check the pixel type, setting the base and multiplier is the responsibility of Cube, not cubeatt.

Reply:
Checking base and multiplier is kind of a proxy for testing that the pixel range was set. I couldn't find a way to retrieve the min/max for the pixel range set besides checking the base and multiplier, since that's what the cube does when you call `Cube.setMinMax()`